haskellPackages: update stackage and hackage

ncfavier opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #296524. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

slotThe committed:
haskellPackages: add slotThe as maintainer for kmonad

I have a commit bit for the repo + Hackage, so can react quickly.

ncfavier committed:
haskellPackages: stackage LTS 22.13 -> LTS 22.14

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

ncfavier committed:
all-cabal-hashes: 2024-03-16T22:28:08Z -> 2024-03-31T04:36:22Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

ncfavier committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

eldritch-cookie committed:
haskell.packages.ghc98.aeson: jailbreak

slotThe committed:
haskellPackages.optparse-applicative-cmdline-util: unbreak

github-actions[bot] committed:
Merge master into haskell-updates

vcunat commented:
Let me delay these builds for a couple days, until PR #298548 gets merged (hopefully, in a day or two).

ncfavier committed:
haskellPackages: fix eval

ncfavier committed:
haskellPackages.smtp-mail: remove obsolete patch

slotThe committed:
haskellPackages.kmonad: unbreak

ncfavier committed:
haskellPackages.castagnoli: unbreak

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #300746 from slotThe/kmonad

haskellPackages.kmonad: unbreak

maralorn committed:
Merge pull request #300506 from eldritch-cookie/hls-ghc9.8-2023-03

haskell.packages.ghc98.aeson: jailbreak

maralorn committed:
Merge pull request #297239 from slotThe/optparse-applicative-cmdline-util-unbreak-021

haskellPackages.optparse-applicative-cmdline-util: unbreak

github-actions[bot] committed:
Merge master into haskell-updates

ncfavier committed:
cabal-install: fix

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

mpscholten committed:
haskellPackages.currencies: Unbroken

The build is broken because the package hasn't been updated for text >= 2 yet

mpscholten committed:
haskellPackages.zip: Don't check

Tests fail on macOS right now

https://github.com/mrkkrp/zip/issues/112

maralorn committed:
Merge pull request #302306 from mpscholten/fix-zip

haskellPackages.zip: Don't check

maralorn commented:
Currently only two relevant appearing build errors on the branch:

- [ ] [[📱❌]](https://hydra.nixos.org/build/254958066) [[🐧❌]](https://hydra.nixos.org/build/254964054) [carp](https://hydra.nixos.org/eval/1805483?filter=carp) @jluttine
- [ ] [[📱❌]](https://hydra.nixos.org/build/254945690) [[🐧❌]](https://hydra.nixos.org/build/254969151) [haskellPackages.eventlog2html](https://hydra.nixos.org/eval/1805483?filter=haskellPackages.eventlog2html) @maralorn

I’ll look into the eventlog thing, not immediately sure what to do about carp.

We also have an eval error on darwin: 
```
in job ‘haskell.compiler.ghc963Binary.x86_64-darwin’:
error: Failed to evaluate ghc-binary-9.6.3: «broken»: is marked as broken
```
What’s going on there?

github-actions[bot] committed:
Merge master into haskell-updates

maralorn commented:
@jluttine So, would it be okay for you if we marked Carp broken for the time being?

maralorn committed:
Merge pull request #302302 from mpscholten/fix-currencies

haskellPackages.currencies: Unbroken

jluttine committed:
carp: mark broken

jluttine commented:
> @jluttine So, would it be okay for you if we marked Carp broken for the time being?

Sure! Here's a PR: https://github.com/NixOS/nixpkgs/pull/302534

wineee committed:
Merge pull request #302534 from jluttine/haskell-updates

carp: mark broken

github-actions[bot] committed:
Merge master into haskell-updates

sgrb committed:
pkgsCross.ghcjs.ghc: make EM_CACHE absolute

emscripten-3.1.51 fails to compile code (with cache locking problems) if
EM_CACHE is relative.

maralorn committed:
haskellPackages.eventlog2html: Fix build

maralorn committed:
carp: Disable build on haskell-updates

maralorn committed:
Merge branch 'master' into haskell-updates
